bin/fuel-core/src/cli/run.rs

```diff
@@ -306,25 +311,53 @@ impl Command {
             info!("Block production disabled");
         }
 
-        let consensus_key = load_consensus_key(consensus_key)?;
-        if consensus_key.is_some() && trigger == Trigger::Never {
+        let consensus_signer = if consensus_aws_kms {
+            let config = aws_config::load_from_env().await;
+            let client = aws_sdk_kms::Client::new(&config);
+
+            // Find if we have a key already
+            let key_id = client.list_keys().send().await?.keys.and_then(|keys| {
+                keys.first()
+                    .map(|key| key.key_id.clone().expect("Missing key_id"))
+            });
```

Comment:
You won't have permission to do this in a live deployment (but works fine for localstack), this code should accept a `key_id` value as an environment variable (probably called `CONSENSUS_KEY_KMS_ARN` or something similar) and use that.

You'll still want to verify that if that env var is populated, the key can be accessed, so probably best to call GetPublicKey (https://docs.rs/rusoto_kms/latest/rusoto_kms/struct.GetPublicKeyRequest.html) with the provided ARN to make sure permissions are good in the live env.

Reply:
Thanks for the heads-up. FIxed in 2c50c4be4.